lib/Conversion/LWEToPolynomial/LWEToPolynomial.cpp

```diff
@@ -116,6 +124,177 @@ struct ConvertRLWEDecrypt : public OpConversionPattern<RLWEDecryptOp> {
+                [](auto attr) -> int { return attr.getCleartextStart(); })
+            .Case<lwe::UnspecifiedBitFieldEncodingAttr>([](auto attr) -> int {
+              llvm_unreachable(
+                  "Upsecified Bit Field Encoding Attribute for cleartext "
+                  "start");
+              return -1;
+            })
+            .Default([](Attribute attr) -> int {
+              llvm_unreachable(
+                  "Unsupported encoding attribute for cleartext start");
+              return -1;
+            });
+
+    // Should return failure if cleartextBitwidth or cleartextStart fail.
+    if (cleartextBitwidth == -1 || cleartextStart == -1) {
+      return failure();
+    }
+
+    // Check that cleartext_start = cleartext_bitwidth for BGV encryption.
+    if (cleartextBitwidth != cleartextStart) {
+      // TODO: Add support for other encryption schemes besides BGV. Left as
+      // future work.
+      op.emitError() << "`lwe.relwe_encrypt` expects BGV encryption"
+                     << " with cleartext_start = cleartext_bitwidth, but"
+                     << " found cleartext_start = " << cleartextStart
+                     << " and cleartext_bitwidth = " << cleartextBitwidth
+                     << ".";
+      return failure();
+    }
+
+    auto isPublicKey =
+        llvm::TypeSwitch<Type, bool>(op.getKey().getType())
+            .Case<lwe::RLWEPublicKeyType>([](auto key) -> bool { return true; })
+            .Case<lwe::RLWESecretKeyType>(
+                [](auto key) -> bool { return false; })
+            .Default([](Type key) -> bool {
+              llvm_unreachable(
+                  "Unsupported key type: Neither public key nor secret key");
+              return false;
+            });
+
+    ImplicitLocOpBuilder builder(loc, rewriter);
+    if (isPublicKey) {
+      auto index0 = builder.create<arith::ConstantIndexOp>(0);
+      tensor::ExtractOp publicKey0 =
+          builder.create<tensor::ExtractOp>(key, ValueRange{index0});
+      auto index1 = builder.create<arith::ConstantIndexOp>(1);
+      tensor::ExtractOp publicKey1 =
+          builder.create<tensor::ExtractOp>(key, ValueRange{index1});
+
+      auto dimension =
+          inputT.getRing().getPolynomialModulus().getPolynomial().getDegree();
+
+      auto elementType = rewriter.getIntegerType(inputT.getRing()
+                                                     .getCoefficientModulus()
+                                                     .getType()
+                                                     .getIntOrFloatBitWidth());
+
+      auto tensorParams = RankedTensorType::get({dimension}, elementType);
+
+      // constantT is 2**(cleartextBitwidth), and is used for scalar
+      // multiplication.
+      auto constantT = builder.create<arith::ConstantOp>(
+          builder.getI32IntegerAttr(1 << cleartextBitwidth));
+
+      // TODO: Add pass options to change the seed (which is currently hardcoded
```

Comment:
Could you please add this and the stddev to an issue about randomness choices for lowering LWEToPolynomial?